jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ScanningAppProvider.java

```diff
@@ -339,7 +339,11 @@ public void setDeploymentManager(DeploymentManager deploymentManager)
     public void setMonitoredResources(List<Resource> resources)
     {
         _monitored.clear();
-        _monitored.addAll(resources);
+        for (Resource resource: resources)
+        {
+            if (resource != null)
+                _monitored.addAll(resources);
+        }
```

Comment:
You need to protect again null resources:
```suggestion
        if (resources != null)
            resources.stream().filter(Objects::nonNull).forEach(_monitored::add);
```
or maybe:

```suggestion
        Optional.ofNullable(resources).stream().flatMap(List::stream)
            .filter(Objects::nonNull)
            .forEach(_monitored::add);
```